File: 1835-find-xor-sum-of-all-pairs-bitwise-and/1835-find-xor-sum-of-all-pairs-bitwise-and.cpp
class Solution {
public:
    int getXORSum(vector<int>& arr1, vector<int>& arr2) {
        int n = arr2.size();
        vector<int> ones(32);
        
        for(int b : arr2) {
            for(int i = 0; i < 32; ++i)
                ones[i] += ((b >> i) & 1);
        }
        
        int ret = 0;
        
        for(int a : arr1) {
            int ans = 0;
            for(int i = 0; i < 32; ++i) {
                if(((a >> i) & 1) and (ones[i] & 1))
                ans += (1 << i);
            }
            
            ret ^= ans;
        }
        
        return ret;
        
        
        
    }
};

File: 1835-find-xor-sum-of-all-pairs-bitwise-and/1835-find-xor-sum-of-all-pairs-bitwise-and.js
/**
 * @param {number[]} arr1
 * @param {number[]} arr2
 * @return {number}
 */
var getXORSum = function(arr1, arr2) {
    let ora = 0;
    let orb = 0;
    for(let a of arr1) {
        ora ^= a;
    }
    for(let b of arr2) {
        orb ^= b;
    }
    return ora & orb;
};